final version of post on CoMM (multimodal representation learning) (#218)
creatis-myriad/creatis-myriad.github.io: collections/_posts/2025-07-09-CoMM.md

---
layout: review
title: "What to align in multimodal contrastive learning?"
tags: representation-learning multimodal deep-learning
author: "Robin Trombetta"
cite:
    authors: "Benoit Dufumier, Javiera Castillo Navarro1, Devis Tuia, Jean-Philippe Thiran"
    title: "What to align in multimodal contrastive learning?"
    venue: "International Conference on Learning Representations (ICLR) 2025"
pdf: "https://openreview.net/pdf?id=Pe3AxLq6Wf"
---

# Highlights

* MMultimodal representation learning is often done by aligning the representation of the unimodals views. The underpinned hypothesis is that each modality is a different view of the same entity.

> * However, in many real world scenarios, the interactions between the views can be more complex, as sometimes the relevant information for a target task can be found only in one modality or, even more complex, as the fusion of complementay information in all the modalities.

Reviewer: ```suggestion
* However, in many real world scenarios, the interactions between the views can be more complex, as sometimes the relevant information for a target task can be found only in one modality or, even more complex, as the fusion of complementary information in all the modalities.
```

$$
I(X_1,X_2;Y) = R + S + U_1 + U_2
$$

where $$R$$ represents the redundant, or *shared*, information, $$S$$ the complementary information and $$U_1$$ (*resp.* $$U_2$$) the information only present in modality 1 (*resp.* modality 2). There we can also decompose the following terms:

$$
I(X_1;Y) = R + U_1 \\
I(X_2;Y) = R + U_2 \\
I(X_1;X_2;Y) = R - S
$$

Most previous multimodal representation learning methods are limited to capturing redundant information R. A recent work introduced FactorCL which can integrate uniquess and redundancy druring the multimodal contrastive learning, but it can not capture synergy, and relies on task-specific information preserving augmentations.

Let consider a set of multimodal augmentations $$T^{*}$$ such that for any $$t \in T^{*}$$ and $$X' = t(X)$$ we have $$I(X,X') = I(X,Y)$$. Let $$f_{\theta}$$ be a neural network parametrized by $$\theta$$ and $$Z_{\theta} = f_{\theta}(X) = f_{\theta}(X_1,X_2)$$ (*resp.* $$Z'_{\theta}$$) the common multimodal representation from the multimodal input $$X$$ (*resp.* $$X'$$).

> * Optimizing the term $$I(Z_{\theta};Z'_{\theta})$$ allows to learn the overall mutual information for the task $$Y$$ $$I(X:Y) = R + S + U_1 + U_2$$

Author: It should be ok now, see the dedicated commit

> * In we set $$T=\{ t_i \}$$, *i.e.* we apply augmentations only to one modality, then optimizing $$I(Z_{\theta}, Z'_{\theta})$$ allows to learn $$R + U_i$$.

Reviewer: Missing word at the beginning

&nbsp;

#### Model architecture and training

The architecture and training procedure proposed by the authors is quite simple, and goes as follows (Figure 2). 

<div style="text-align:center">
<img src="/collections/images/CoMM/comm_model.jpg" width=800></div>

> <p style="text-align: center;font-style:italic">Figure 2. (left) CoMM's model architecture. Inputs from different modalities are encoded by modality-specific encoders. Modality-specific features are processed by atent converters to map them into sequences of embeddings which are concatenated and fused by a transformer block. (right) CoMM training for two modalities. Two multimodal augmentations are applied to X to obtain X' and X'', and each modality is also passed through its associated projection operator. The n+2 transformed versions of X are processed by the network and trained to maximize their agreement using contrastive objectives.</p>

Reviewer: ```suggestion
<p style="text-align: center;font-style:italic">Figure 2. (left) CoMM's model architecture. Inputs from different modalities are encoded by modality-specific encoders. Modality-specific features are processed by latent converters to map them into sequences of embeddings which are concatenated and fused by a transformer block. (right) CoMM training for two modalities. Two multimodal augmentations are applied to X to obtain X' and X'', and each modality is also passed through its associated projection operator. The n+2 transformed versions of X are processed by the network and trained to maximize their agreement using contrastive objectives.</p>
```

> * Each modality is ended independently by its specific encoder

Reviewer: ```suggestion
* Each modality is encoded independently by its specific encoder
```

* Linear modules transform features into modality-specific sequences of embeddings
* The embeddings of all modalities are concatenated and fused by self-attention layers to obtain the final multimodal embedding $$Z$$.

> Given multimodal input $$X=(X_1,...,X_n)$$ and a set of label-preserving multimodal transforms $$T^{*}$$, two random augmentations $$t',t''$$ are drawn from $$T^{*}$$ and applied to $$X$$ to obtain $$X' = t'(X)$$ and $$X'' = t''(X)$$. We also consider the projections $$X_i = ( [MASK],...,X_i,...,[MASK] )$$ for $$i \in \{ 1,...n, \}$$ which mask every modality except for the $$i$$-th. For one multimodal input, it gives $$(n+2)$$ transformed verions of the input, which we all encode by th network to obtain their corresponding embeddings : $$Z'$$, $$Z''$$ and $$\{ Z_i\}_{i=1}^{n}$$. As briefly discussed before, several terms are optimized during the training : $$I(Z', Z'')$$ to maximize $$R + S + \sum_{i} U_i$$ and $$I(Z_i,Z')$$ and $$I(Z_i,Z'')$$ to better approximate $$R + U_i$$. The authors used the InfoNCE loss to optimize their network:

Reviewer: ```suggestion
Given multimodal input $$X=(X_1,...,X_n)$$ and a set of label-preserving multimodal transforms $$T^{*}$$, two random augmentations $$t',t''$$ are drawn from $$T^{*}$$ and applied to $$X$$ to obtain $$X' = t'(X)$$ and $$X'' = t''(X)$$. We also consider the projections $$X_i = ( [MASK],...,X_i,...,[MASK] )$$ for $$i \in \{ 1,...n, \}$$ which mask every modality except for the $$i$$-th. For one multimodal input, it gives $$(n+2)$$ transformed versions of the input, which we all encode by the network to obtain their corresponding embeddings : $$Z'$$, $$Z''$$ and $$\{ Z_i\}_{i=1}^{n}$$. As briefly discussed before, several terms are optimized during the training : $$I(Z', Z'')$$ to maximize $$R + S + \sum_{i} U_i$$ and $$I(Z_i,Z')$$ and $$I(Z_i,Z'')$$ to better approximate $$R + U_i$$. The authors used the InfoNCE loss to optimize their network:
```

$$
\hat{I}_{\textrm{NCE}}(Z,Z') = \mathbb{E}_{z,z'_{pos},z'_{neg}} \left[ \log \frac{\textrm{exp sim}(z,z'_{pos})}{\sum_{z'_{neg}} \textrm{exp sim}(z,z'_{neg})} \right]
$$

The overall optimization objective is the following: 

$$
\mathcal{L}_{\textrm{CoMM}} = - \underbrace{\hat{I}_{\textrm{NCE}}(Z',Z'')}_{\approx R + S + \sum_{i=1}^{n}U_i} - \sum_{i=1}^{n} \underbrace{\frac{1}{2} \left( \hat{I}_{\textrm{NCE}}(Z_i,Z') + \hat{I}_{\textrm{NCE}}(Z_i,Z'')  \right)}_{\approx R+U_i} =: \mathcal{L} + \sum_{i=1}^{n}\mathcal{L}_i
$$

&nbsp;

# Experiments

> The authors conducted two main sets of experiments. First, they used a synthetic dataset to define a controlled environment and investigate the benefits of their model for capturing redundancy, uniquess and synergy. Second, they evaluate their self-supersied learning method on various benchmarks (see Figure 3 for examples). 

Reviewer: ```suggestion
The authors conducted two main sets of experiments. First, they used a synthetic dataset to define a controlled environment and investigate the benefits of their model for capturing redundancy, uniqueness and synergy. Second, they evaluate their self-supervised learning method on various benchmarks (see Figure 3 for examples). 
```

#### Synthetic bimodal trifeature dataset

> The trifeature dataset contains images of one of ten shapes (round, square, star, etc.), with one of ten textures (dots, stripes, etc.) and one of ten colors, for a total of 1 000 combinations. A fake bimodal dataset is created by given taking two different images from the Trifeature dataset. 

Reviewer: ```suggestion
The trifeature dataset contains images of one of ten shapes (round, square, star, etc.), with one of ten textures (dots, stripes, etc.) and one of ten colors, for a total of 1 000 combinations. A fake bimodal dataset is created by taking two different images from the Trifeature dataset. 
```

<div style="text-align:center">
<img src="/collections/images/CoMM/example_trifeature.jpg" width=300></div>
<p style="text-align: center;font-style:italic">Figure 3. Examples of images sampled from the Trifeature dataset.</p>

> ***Redundancy.*** To model redundancy, shape is chosen as shared feature between the modalities. It means that everytime each time a pair is sampled and given to the bimodal neural network, the first image and the second image always have the same shape. The ability of the latent representation to capture this redundant feature is evaluated by the linear probing accuracy of shape prediction (change level of 10%). Training and test sets follow the same distribution with respectively 10 000 and 4096 images.

Reviewer: ```suggestion
***Redundancy.*** To model redundancy, shape is chosen as shared feature between the modalities. It means that each time a pair is sampled and given to the bimodal neural network, the first image and the second image always have the same shape. The ability of the latent representation to capture this redundant feature is evaluated by the linear probing accuracy of shape prediction (change level of 10%). Training and test sets follow the same distribution with respectively 10 000 and 4096 images.
```

> ***Uniquenss.*** Texture is chosen as the unique feature in the bimodal dataset. For a given pair of images (with similar shape) which form the bimodal input, the two images have different textures. The capacity of the model and the output multimodal latent representation $$Z$$ to capture unique information is measured by the linear probing accuracy to predict the texture of the first image. The same metric is measure with the texture of the second image (chance level of 10%). Training and test sets follow the same distribution with respectively 10 000 and 4096 images.

Reviewer: ```suggestion
***Uniqueness.*** Texture is chosen as the unique feature in the bimodal dataset. For a given pair of images (with similar shape) which form the bimodal input, the two images have different textures. The capacity of the model and the output multimodal latent representation $$Z$$ to capture unique information is measured by the linear probing accuracy to predict the texture of the first image. The same metric is measure with the texture of the second image (chance level of 10%). Training and test sets follow the same distribution with respectively 10 000 and 4096 images.
```

***Synergy***. Modelling and measuring synergy is trickier. To do so, the authors have introduced bias in the training set between the two modalities. In the two previous experiments, the color and texture of the two images and drawn independently. In this experiment, they define a mapping $$\mathcal{M}$$ between the ten textures and ten colors. They randomly selected 10 000 pairs of images respecting this mapping for the training. The test set is the same as above, and the task is here to predict whether an image satisfies the mapping $$\mathcal{M}$$ or not (change level of 50%).

&nbsp;

> The performance on the three tasks are shown in Figure 4. While redundacy information is easily to captured by all methods, uniqueness can not be seen with standard multimodal alignment. CoMM is the onlyy method to accurately capture synergyzing information between two modalities.

Reviewer: ```suggestion
The performance on the three tasks are shown in Figure 4. While redundacy information is easily captured by all methods, uniqueness can not be seen with standard multimodal alignment. CoMM is the only method to accurately capture synergyzing information between two modalities.
```
